pyro/ops/stats.py

```diff
@@ -397,3 +397,44 @@ def fit_generalized_pareto(X):
     k = k * N / (N + 10.0) + 5.0 / (N + 10.0)
 
     return k, sigma
+
+
+def crps_empirical(pred, truth):
+    """
+    Computes negative Continuous Ranked Probability Score CRPS* [1] between a
+    set of samples ``pred`` and true data ``truth``. This uses an ``n log(n)``
+    time algorithm to compute a quantity equal that would naively have
+    complexity quadratic in the number of samples ``n``::
+
+        CRPS* = E|pred - truth| - 1/2 E|pred - pred'|
+              = (pred - truth).abs().mean(0)
+              - (pred - pred.unsqueeze(1)).abs().mean([0, 1]) / 2
+
+    Note that for a single sample this reduces to absolute error.
+
+    References
+    [1] `Strictly Proper Scoring Rules, Prediction, and Estimation`
+    Tilmann Gneiting, Adrian E. Raftery (2007)
+    https://www.stat.washington.edu/raftery/Research/PDF/Gneiting2007jasa.pdf
+
+    :param torch.Tensor pred: A set of sample predictions batched on rightmost dim.
+        This should have shape ``(num_samples,) + truth.shape``.
+    :param torch.Tensor truth: A true tensor.
```

Comment:
is there a better description than 'true tensor'?  tensor of observations or the like?

Reply:
done